src/cpp/session/modules/SessionPythonEnvironments.R

```diff
@@ -43,11 +43,92 @@
    .rs.python.describeInterpreter(pythonPath)
 })
 
+.rs.addFunction("python.findCondaForEnvironment", function(envPath)
+{
+   # locate history file
+   historyFile <- file.path(envPath, "conda-meta/history")
+   if (!file.exists(historyFile))
+      return("")
+   
+   # read first few lines
+   contents <- readLines(historyFile, n = 2L, warn = FALSE)
+   if (length(contents) < 2L)
+    return("")
+
+   # parse conda path
+   line <- substring(contents[2L], 8L)
+   index <- regexpr(" ", line, fixed = TRUE)
+   if (index == -1L)
+      return("")
+
+   conda <- substring(line, 1L, index - 1L)
+   if (.rs.platform.isWindows)
+      conda <- file.path(dirname(conda), "conda.exe")
+
+   # prefer condabin if it exists
+   condabin <- file.path(dirname(conda), "../condabin", basename(conda))
+   if (file.exists(condabin))
+      conda <- condabin
+   
+   # bail if conda wasn't found
+   if (!file.exists(conda))
+      return("")
+   
+   .rs.canonicalizePath(conda)
+  
+})
+
+.rs.addFunction("python.configuredInterpreterPath", function(projectDir)
+{
+   # check some pre-defined environment variables
+   vars <- c("RENV_PYTHON", "RETICULATE_PYTHON")
+   for (var in vars) {
+      value <- Sys.getenv(var, unset = NA)
+      if (!is.na(value))
+         return(value)
+   }
+   
+   # if this project has a local interpreter, use it
+   if (!is.null(projectDir))
+   {
+      projectPython <- .rs.python.projectInterpreterPath(projectDir)
+      if (file.exists(projectPython))
+         return(projectPython)
+   }
+   
+   # check version of Python configured by user
+   prefsPython <- .rs.readUiPref("python_path")
```

Comment:
Quick reminder (if it isn't already resolved) to make sure that we persist python_path that is a subdirectory of the project as a relative path.

Reply:
Should be fixed up now -- thanks for spotting that.